mesh/nurbs.cpp

```diff
@@ -45,6 +45,53 @@ KnotVector::KnotVector(int order, int NCP)
    knot = -1.;
 }
 
```

Comment:
I like this constructor; the `repeated` check is a nice safeguard but it also means this will often initialize a `KnotVector` with a vector different from what was provided. IMO adding either a warning or even just a small comment in the header (I think that is sufficient) would be helpful.